### ISSUE TITLE
Connecting the ADXL345 accelerometer to the Raspberry Pi 3

### ISSUE THREAD

**Issue by ccamacho** [pubstack/pubstack.github.io#3]

**Comment by bretcj7** [comment]
Great blog post, worked effortlessly and no issues.  Thank you.

**Comment by ccamacho** [comment]
Awesome you like it!

**Comment by pimyride94** [comment]
Just a question before I choose which accelerometer to buy. The ADXL345 isn't an analog accelerometer? That way don't I need an A/D converter for it to work on my raspberry pi 3 model B?

**Comment by bretcj7** [comment]
Its digital.  Works with  I2C or SPI  just hook up on the correct GPIO pins.
http://www.analog.com/media/en/technical-documentation/data-sheets/ADXL345.pdf

**Comment by pimyride94** [comment]
Oh, my mistake. That's good then! Thanks

2017-07-06 16:48 GMT-03:00 Bret Askeland <notifications@github.com>:

**Comment by Haroldov** [comment]
Hola, seguiste trabajando con este acelerometro? Vi que pusiste que el siguiente blog del acelereometro seria trabajarlo en tiempo real pero no he podido encontrar ese blog. Quitame la duda y muchas gracias!

**Comment by pimyride94** [comment]
Hey, I have been working with It. I might be able to help you, what do you
want to know?

Em 26 de out de 2017 3:17 AM, "Haroldo Vélez" <notifications@github.com>
escreveu:

**Comment by ccamacho** [comment]
Yeah, I did some more research about it but didn't have the chance to write all down. @pimyride94 @Haroldov we can sync if you want to write about it.

**Comment by kacunar** [comment]
@ccamacho @pimyride94  i want to know  what is  the use of the BW_RATE in the class code.

**Comment by ccamacho** [comment]
Its how frequent you will get the accelerometer measurements. More info here: https://www.dimensionengineering.com/info/accelerometers

**Comment by kacunar** [comment]
@ccamacho yes but I couldn't implement this rate in the code. Can you help me with this?

**Comment by ccamacho** [comment]
@kacunar those values represent registers rate codes, you can check the accelerometer specification at https://www.sparkfun.com/datasheets/Sensors/Accelerometer/ADXL345.pdf

In any case, I'll give you an example, the 1600HZ rate (in the code) corresponds with a 1111 rate code (base 2), which is what you have in the code 0x0F (base 16).

**Comment by kacunar** [comment]
@ccamacho yes I know. But when I run my code, I only get one measurement. why the accelerometer dont give me measurements with this frecuency??

**Comment by pimyride94** [comment]
Well, you have tô make a code tô give you multiple readings, such as a
"while" iteration. For example:
 while true:
     x, y, z = Get.adxl345.readings
     time.sleep(0.1)

This Will give you the Axis values each 0.1 Second. The rate you specify is
the ability tô read and not How many times It Will give you a value per
second
Hope this helps



Em 27 de out de 2017 8:15 PM, "Kevin Acuña" <notifications@github.com>
escreveu:

**Comment by kacunar** [comment]
@pimyride94 ohhh ok, thank you. This is how i have the code now. But i have errors with the rate because the code is large. So when  I want 10 seconds of measure with a sampling frecuency of 1 HZ , the raspberry give me  10 measurements but it takes more than 10 seconds.

**Comment by pimyride94** [comment]
So maybe you Will need tô separate the while iteration from the rest of the
code. Make the while and create a variable tô count, like "i". So each time
a while loop occurs the variable increases by one. Then inside the
iteration you ask, is i < 10, If thats true use break, and It Will get out
of the while loop. Also inside the iteration you can store the readings in
an array tô work with them as you wish later on the code and It Will last
10 second tô get all 10 readings. But as you Said, the rest of the code
Will take more time to complete

Em 27 de out de 2017 9:04 PM, "Kevin Acuña" <notifications@github.com>
escreveu:

**Comment by kacunar** [comment]
@pimyride94 thank you
